### DISCUSSION TITLE
Problems with the detection

### DISCUSSION THREAD

**Discussion by Bongo2** [Nachtzuster/BirdNET-Pi#257] [General]
I have problems with the detection.
With most species I am sure that it is not possible.
![aaaaa](https://github.com/user-attachments/assets/ecc54ad3-3286-4c81-af05-eb2122794baa)

This is unsatisfactory.

If you look at the spectogram, you can only see noise. 
![hc_2963](https://github.com/user-attachments/assets/20febd61-d204-4e9b-9a9c-892d8a10ab7f)

But the detection is 82%. This is how many spectograms look. That can't be. Do I do something wrong or is that normal? How can I improve detection?

My settings:
Audio about RTSP Stream of my surveillance camera.
![hc_2961](https://github.com/user-attachments/assets/19287580-86ea-48bf-acf6-b78095c9b7fc)
![hc_2962](https://github.com/user-attachments/assets/e2808801-e4ed-47df-a089-626431f61735)

**Comment by tphakala** [comment]
This is a problem with BirdNET v2.4 model, it is detecting porzana porzana also at my location from simple silence. There are few other species also which have clearly been trained with poor source data causing common false detections. Just add that species to ignore list.

Feedback of BirdNET model should be directed at BirdNET Analyzer repository as this is source of AI model, however they are quite unresponsive. https://github.com/kahst/BirdNET-Analyzer/issues/512

**Reply by alexbelgium** [reply]
Indeed... I have also a list of species that are erroneous due to various noises. In the end I simply exclude them as you describe tphakala...

**Comment by alexbelgium** [comment]
> I have problems with the detection. With most species I am sure that it is not possible. ![aaaaa](https://private-user-images.githubusercontent.com/45128074/395790009-ecc54ad3-3286-4c81-af05-eb2122794baa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUzMzcxMTQsIm5iZiI6MTczNTMzNjgxNCwicGF0aCI6Ii80NTEyODA3NC8zOTU3OTAwMDktZWNjNTRhZDMtMzI4Ni00YzgxLWFmMDUtZWIyMTIyNzk0YmFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI3VDIyMDAxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk3YTkzN2YxMzQxMmZjNjFiNzYwY2VlMGVmZWI1NDQ1MGQ0MmZiNTJjM2QxOWNiYjEzYWVjOTdmMjUwZDRhODUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.j4nFpKHmPqsSov3799r5DUv_P5IBG4k85-3NqOMXhxA)
> 
> This is unsatisfactory.
> 
> If you look at the spectogram, you can only see noise. ![hc_2963](https://private-user-images.githubusercontent.com/45128074/395790058-20febd61-d204-4e9b-9a9c-892d8a10ab7f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUzMzcxMTQsIm5iZiI6MTczNTMzNjgxNCwicGF0aCI6Ii80NTEyODA3NC8zOTU3OTAwNTgtMjBmZWJkNjEtZDIwNC00ZTliLTlhOWMtODkyZDhhMTBhYjdmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI3VDIyMDAxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ5NzkxZTU2MDVkZjk3MDdkYzc4MTI3MGViMjI4NWJhNTZjNGU5YjY3MzM5MWExMzA3Yzk0MzcxMTFhMmU1ZjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.nBcd7_t5ynMLssApRm_tRV6JKvWEfs-nHCm6FqyvDsw)
> 
> But the detection is 82%. This is how many spectograms look. That can't be. Do I do something wrong or is that normal? How can I improve detection?
> 
> My settings: Audio about RTSP Stream of my surveillance camera. ![hc_2961](https://private-user-images.githubusercontent.com/45128074/395790094-19287580-86ea-48bf-acf6-b78095c9b7fc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUzMzcxMTQsIm5iZiI6MTczNTMzNjgxNCwicGF0aCI6Ii80NTEyODA3NC8zOTU3OTAwOTQtMTkyODc1ODAtODZlYS00OGJmLWFjZjYtYjc4MDk1YzliN2ZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI3VDIyMDAxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZiMjA1NWZkZjU1NGI2MTBhOWZhNzg2ZWU4ZjNmYjIyODA5Y2E0OGMwOTI4ZjdiNDVkNDVhYzlmY2I0Yjc0YjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.iD7Mp88AnhsHdIv2oVgVjSp_FYm0MPdTsYjjUVi10Dw) ![hc_2962](https://private-user-images.githubusercontent.com/45128074/395790103-e2808801-e4ed-47df-a089-626431f61735.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUzMzcxMTQsIm5iZiI6MTczNTMzNjgxNCwicGF0aCI6Ii80NTEyODA3NC8zOTU3OTAxMDMtZTI4MDg4MDEtZTRlZC00N2RmLWEwODktNjI2NDMxZjYxNzM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI3VDIyMDAxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJjZWRkODU1MDc1ZThlMjExYjIyNzVjYThiYWMzNDM0ODhjMTg1MTVjMjNjYjU1NmVhZGJlYzc0YTIwNjU5YTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jpBxQpeaAOa-Ptv8ZChnbg6y-cUp4IcTXSxgTuGQyrQ)

You could try by un checking the species range model 2.4 ; and increasing your gain. The signal could be too flat which might cause the false positives

**Reply by Bongo2** [reply]
Thanks for your answer.

> and increasing your gain. 

Which gain do you mean?

**Reply by alexbelgium** [reply]
Hi, I meant using alsamixer, pulseaudio, or the rtsp stream to adapt the volume of the microphone. Here is a guide using pulseaudio : https://github.com/mcguirepr89/BirdNET-Pi/wiki/Adjusting-your-sound-card

Globally, it increases the sound volume of your microphone. You need to select a setting that allows to show a good volume, while avoiding clipping and too much noise which would drown lesser signals

**Comment by Bongo2** [comment]
Do I have to leave the sound card to standard? 
The controller for recording is 100. Should I make it less?
![hc_3055](https://github.com/user-attachments/assets/13a4e5f5-b6fd-4597-a4dc-8bcbee109b87)

**Reply by alexbelgium** [reply]
Edit : removed my comment which was erroneous about what meant red color in alsamixer (see tphakala comment below)

For the playing setting you can have it at 0 even, only the recording part is used.

If useful I put some tips that I learned while setting my system here : https://github.com/alexbelgium/hassio-addons/blob/master/birdnet-pi/DOCS.md

**Reply by tphakala** [reply]
alsamixer does not show input levels, so red does not indicate clipping in here. Most sound cards produce very quiet input volumes so I would always set capture level to 100 here.

**Reply by alexbelgium** [reply]
Thanks for the feedback! I'm using the focusrite scarlet 2i2 that you recommended to me and need to use something like 40% (+30dB) to have a signal of a good quality with my em272 and avoid losing precision in the section below 500hz. I was initially using 50%. This is a rabbit hole for this thread and I apologize but what is the gain level that you use with your AOM5024 system?

**Reply by tphakala** [reply]
I have set levels on alsamixer to 100 and I set gain on audio interface to about 75%. I have not done any scientific measurements on frequency response, but I have noticed that there are some variations on low end frequency response between some AOM5024 capsules.

**Reply by blogdemoi** [reply]
> I have set levels on alsamixer to 100 and I set gain on audio interface to about 75%. I have not done any scientific measurements on frequency response, but I have noticed that there are some variations on low end frequency response between some AOM5024 capsules.

May I ask what audio interface  you are using?

**Reply by tphakala** [reply]
I have few audio interfaces, Focusrite Scarlett 2i2 3rd gen, Scarlett Solo 4th gen, Zoom AMS-24 and M-Audio m-track solo. Of these only Focusrites are in active use at the moment. 4th Gen Solo is used for this https://www.youtube.com/watch?v=889jH2RUxI8

**Reply by blogdemoi** [reply]
Thanks. I have been running a BirdnetPi for a long time now with a EM272, but  I would like to amplify the sound a little, what kind of audio interface would you recommend? That I can use 24/7?

**Reply by tphakala** [reply]
Focusrite Scarlett interfaces are very good, works well on Linux and has good preamps. From current lineup 4th Gen Solo is maybe best budget option. Is your EM272 with XLR connection and phantom power input?

**Reply by blogdemoi** [reply]
My EM272 is connected to a simple usb Ugreen interface, no XLR connection, no phantom power. It's a simple setup.

**Comment by Bongo2** [comment]
@alexbelgium 
I changed the recording from 100 to 50.
I think the detection is better. There are not so many false detections anymore.

@tphakala 
I have questions about that.
You have an audio interface "Scarlett solo 4th gen" where the microphone is connected. How is it connected to the Raspberry? With USB?

I also have several surveillance cameras in nesting box and bird feeding. I use the audio stream from this camera.
https://www.youtube.com/watch?v=ePVXd7whTFY

It is an easy way because I don't need a further microphone. But the quality is probably not good enough.

If I see it correctly, the species is shown in your video stream. How did you do that?

And I see you are the author of BirdNet-GO with a docker integration. This is very interesting for me. I have to see that. I have some docker applications on my Synology.
